presto-orc/src/main/java/io/prestosql/orc/reader/StructColumnReader.java

```diff
@@ -96,8 +98,17 @@
             fieldNames.add(fieldName);
 
             OrcColumn fieldStream = nestedColumns.get(fieldName);
+
             if (fieldStream != null) {
-                structFields.put(fieldName, createColumnReader(field.getType(), fieldStream, systemMemoryContext, blockFactory));
+                if (readLayout.isFullyProjected()) {
```

Comment:
Good catch! you're totally right. Earlier I was split on whether, 

when `ProjectedLayout::fieldLayouts` is empty 

(1) have `getFieldLayout` return a `fullyProjectedLayout` OR 
(2) throw an exception, since we may not want to return fullyProjectedLayout for any **random** field name. 

(i.e. invoking `getFieldLayout(z)` for a fully projected layout of `a`, where `a` has only two fields `x` and `y`.) 

But since we only use this method after the `fieldStream != null` check, I think (1) should be okay. 

Removed the check and the `isFullyProjected` method.